### PR TITLE
use rate maps correctly in mutgen

### PR DESCRIPTION
**Pull request by petrelharp**
We forgot a `left = right`. :facepalm:

I've put in some tests and some verification along the lines of what @awohns has in #1470, but we could probably think of one or two more tests (especially having to do with rate maps + discrete mutations).

### CODE REVIEW

**Comment by petrelharp** [comment]
The verification plots:
test_varying_rate
and
test_subdivide

**Comment by petrelharp** [comment]
Ok, added a discrete verification: 
mutation_counts

**Comment by jeromekelleher** [comment]
> We forgot a left = right. ￼

Pretty sure that's my fault - doh! Thanks for tracking this down @awohns and for fixing @petrelharp!

File: tests/test_mutations.py [hunk 1]
```diff
@@ -275,6 +275,45 @@ def test_new_model(self):
         self.validate_stationary(model)
 
 
+class TestMutateRateMap:
+    def test_no_muts(self):
+        ts = msprime.sim_ancestry(10, sequence_length=10, random_seed=2)
```
Comment (jeromekelleher) [inline tests/test_mutations.py:280]: Be good to add one more test like this here,
```python
 def test_many_edges(self):
￼        ts = msprime.sim_ancestry(10, sequence_length=10, random_seed=2, recombination_rate=10)
        pos = [0, 3, 7, 10]
￼        rate = [1.0, 0.0, 1.0]
￼        mut_map = msprime.RateMap(position=pos, rate=rate)
￼        ts = msprime.sim_mutations(
￼            ts, rate=mut_map, random_seed=1, discrete_genome=True
￼        )
￼        site_pos = ts.tables.sites.position
￼        assert np.all(site_pos == [0, 1, 2, 7, 8, 9])
```

File: verification.py [hunk 2]
```diff
@@ -4384,6 +4384,111 @@ def test_arbitrary_model_stats(self):
...
+                            rate_map.start_position, rate_map.end_position, 101
+                        ),
+                    ]
+                )
+            )
+        )
+        sub_rate = [rate_map.get_rate(p) for p in sub_pos[:-1]]
+        sub_map = msprime.RateMap(position=sub_pos, rate=sub_rate)
+        t0 = msprime.sim_mutations(
+            ts, rate=rate_map, random_seed=789, discrete_genome=discrete
```
Comment (jeromekelleher) [inline verification.py:4405]: We shouldn't use random seeds for the verification tests - they should be robust to small random effects, and we want to do different seeds every time.

**Review by jeromekelleher — COMMENTED** [review]
Looks great, thanks @petrelharp. A couple of minor comments - we do want to get rid of the random seeds in the verifiation tests, though.

**Comment by petrelharp** [comment]
Ok - fixed those things up. Haven't squashed so it's easy to see.

I'm particularly happy with the statistical verification here, thanks to `ts.segregating_sites( )` giving us the for-reals expectation. =)

**Comment by petrelharp** [comment]
Ok - fixed those things up. Haven't squashed so it's easy to see.

I'm particularly happy with the statistical verification here, thanks to `ts.segregating_sites( )` giving us the for-reals expectation. =)

**Comment by petrelharp** [comment]
Ok - fixed those things up. Haven't squashed so it's easy to see.

I'm particularly happy with the statistical verification here, thanks to `ts.segregating_sites( )` giving us the for-reals expectation. =)

**Comment by petrelharp** [comment]
Squashed.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**